app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,27 @@
+class RecipesController < ApplicationController
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def create
+    @recipe = current_user.recipes.create(recipe_params)
+    if @recipe.save
+      flash[:notice] = "Recipe saved."
+      redirect_to @recipe
+    else
+      flash[:error] = "There was an error saving your recipe: #{@recipe.errors.full_messages.first}"
+      redirect_to new_recipe_path
```

Comment:
If there are validation errors, the form will lose previously entered data upon redirect I'd expect which might be upsetting for users. 

Consider adding an unhappy path to the existing feature spec to check that an invalid form submission remembers the previously entered data.

Once this unhappy path in the spec is failing as you'd expect, then one possible fix will look like this:

```ruby
# Change redirect_to line from:
redirect_to new_recipe_path
# to:
render :new
```

The redirect triggers a whole new request, meaning the "new" view is rendered with a **blank** `@recipe` that is initialized in the `new` action.

Instead of a redirect, `render :new` will render the `new.html.erb` template using the invalid, but pre-populated @recipe initialized in the `create` action. A new request is not issued and the `new` action is not executed (unlike with a redirect).